coral/src/app/features/requests/components/topics/components/TopicRequestsTable.tsx

```diff
@@ -0,0 +1,148 @@
+import { DataTable, DataTableColumn, EmptyState } from "@aivenio/aquarium";
+import infoIcon from "@aivenio/aquarium/dist/src/icons/infoSign";
+import editIcon from "@aivenio/aquarium/dist/src/icons/edit";
+import deleteIcon from "@aivenio/aquarium/dist/src/icons/delete";
+import { TopicRequest } from "src/domain/topic/topic-types";
+import {
+  requestStatusChipStatusMap,
+  requestStatusNameMap,
+} from "src/app/features/approvals/utils/request-status-helper";
+import {
+  requestOperationTypeChipStatusMap,
+  requestOperationTypeNameMap,
+} from "src/app/features/approvals/utils/request-operation-type-helper";
+
+interface TopicRequestTableRow {
+  id: TopicRequest["topicid"];
+  topicname: TopicRequest["topicname"];
+  environmentName: TopicRequest["environmentName"];
+  requestStatus: TopicRequest["requestStatus"];
+  requestOperationType: TopicRequest["requestOperationType"];
+  teamname: TopicRequest["teamname"];
+  requestor: TopicRequest["requestor"];
+  requesttimestring: TopicRequest["requesttimestring"];
+}
+
+type TopicRequestsTableProps = {
+  requests: TopicRequest[];
+  onDetails: (topicId: string) => void;
+  onEdit: (topicId: string) => void;
+  onDelete: (topicId: string) => void;
+};
+
+function TopicRequestsTable({
+  requests,
+  onDetails,
+  onEdit,
+  onDelete,
+}: TopicRequestsTableProps) {
+  const columns: Array<DataTableColumn<TopicRequestTableRow>> = [
+    { type: "text", field: "topicname", headerName: "Topic" },
+    {
+      type: "status",
+      field: "environmentName",
+      headerName: "Environment",
+      status: ({ environmentName }) => ({
+        status: "neutral",
+        text: environmentName,
+      }),
+    },
+    {
+      type: "status",
+      field: "requestStatus",
+      headerName: "Status",
+      status: ({ requestStatus }) => {
+        return {
+          status: requestStatusChipStatusMap[requestStatus],
+          text: requestStatusNameMap[requestStatus],
+        };
+      },
+    },
+    {
+      type: "status",
+      field: "requestOperationType",
+      headerName: "Type",
+      status: ({ requestOperationType }) => {
+        return {
+          status: requestOperationTypeChipStatusMap[requestOperationType],
+          text: requestOperationTypeNameMap[requestOperationType],
+        };
+      },
+    },
+    { type: "text", field: "teamname", headerName: "Owned by" },
+    { type: "text", field: "requestor", headerName: "Requested by" },
+    {
+      type: "text",
+      field: "requesttimestring",
+      headerName: "Requested on",
+      formatter: (value) => {
+        return `${value}${"\u00A0"}UTC`;
+      },
+    },
+    {
+      type: "action",
+      headerName: "Details",
+      headerInvisible: true,
+      width: 30,
+      action: ({ id }) => ({
+        text: "View",
+        icon: infoIcon,
+        onClick: () => onDetails(String(id)),
+      }),
+    },
+    {
+      type: "action",
+      headerName: "Edit",
```

Comment:
I think for the Edit and Delete actions, we could implement their disabled state in this PR. For example:

```tsx
    {
      type: "action",
      headerName: "Edit",
      headerInvisible: true,
      width: 30,
      action: ({ id, isEditable }) => ({
        text: "Edit",
        icon: editIcon,
        onClick: () => onEdit(String(id)),
        disabled: !isEditable
      }),
    },

```

Reply:
Sure, i can add the disabled check also.

Reply:
Done!